src/linuxdesktop.cc

```diff
@@ -45,17 +45,56 @@ namespace {
   // -----------------------------------------------------------------------------------------------
   QPixmap grabScreenDBusKde()
   {
-    QDBusInterface interface(QStringLiteral("org.kde.KWin"),
-                             QStringLiteral("/Screenshot"),
-                             QStringLiteral("org.kde.kwin.Screenshot"));
-    QDBusReply<QString> reply = interface.call(QStringLiteral("screenshotFullscreen"));
-    QPixmap pm(reply.value());
-    if (!pm.isNull()) {
-      QFile::remove(reply.value());
-    } else {
-      logError(desktop) << LinuxDesktop::tr("Screenshot via KDE DBus interface failed.");
+    // Create a temporary file to receive the screenshot data
+    QTemporaryFile tempFile;
+    if (!tempFile.open()) {
+      qDebug() << "Failed to create temporary file";
```

Comment:
Here and other locations: Please use `logWarning(desktop)` and `logError(deskop)` within this file.
(and additionally tr for strings that should be translatable)
e.g.
```cpp
logError(desktop) << tr("Failed to create temporary file");
```